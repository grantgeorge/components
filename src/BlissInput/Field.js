// @flow

import styled from 'styled-components'

export const Field = styled.div`
  padding: 0;
  background: transparent;
  border: 0;
  color: #212529;
  outline: none;
  cursor: text;
  display: block;
  width: 100%;
  line-height: 32px;
  padding-bottom: 3px;
  transition: opacity 200ms ease-in-out;
  font-weight: 400;

  &:disabled: {
    cursor: not-allowed;
    color: rgba(119, 119, 119, 0.65);
  }

  ::-webkit-input-placeholder {
    color: #8898aa;
  }
  ::-moz-placeholder {
    color: #8898aa;
  }
  ::-ms-input-placeholder {
    color: #8898aa;
  }

  /* IE doesn't show placeholders when empty+focused */
  :-ms-input-placeholder {
    color: #fff;
  }

  ::placeholder {
    color: #8898aa;
  }

  &.is-empty:not(.is-focused) {
    opacity: 0;
  }

  &.is-focused + span span,
  &:not(.is-empty) + span span {
    transform: scale(0.68) translateY(-36px);
    cursor: default;
  }

  &.is-focused + span span,
  &:not(.is-empty) + span span {
    transform: scale(0.68) translateY(-36px);
    cursor: default;
  }

  &.is-focused + span {
    border-bottom-color: #007bff;
  }
`

export default Field
