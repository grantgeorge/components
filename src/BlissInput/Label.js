// @flow

import styled from 'styled-components'

export const Label = styled.label`
  height: 35px;
  position: relative;
  color: #8798ab;
  display: block;
  margin: 0 0 1.75rem;
`

export default Label
