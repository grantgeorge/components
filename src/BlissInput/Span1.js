// @flow

import styled from 'styled-components'

import Input from './Input'

// border-bottom-color: #007bff;

export const Span1 = styled.span`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  line-height: 32px;
  color: #8798ab;
  border-bottom: 1px solid #586a82;
  transition: border-bottom-color 200ms ease-in-out;
  cursor: text;
  pointer-events: none;

  ${Input}.is-focused {
    border-bottom-color: red;
  }
`

export default Span1
