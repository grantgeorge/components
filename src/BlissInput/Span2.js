// @flow

import styled from 'styled-components'

export const Span2 = styled.span`
  position: absolute;
  top: 0;
  left: 0;
  transform-origin: 0% 50%;
  transition: transform 200ms ease-in-out;
  cursor: text;
  font-weight: 400;
`

export default Span2
