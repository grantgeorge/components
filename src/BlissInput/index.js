// @flow

import React, { PureComponent } from 'react'
import classNames from 'classnames'

import Input from './Input'
import Label from './Label'
import Span1 from './Span1'
import Span2 from './Span2'

export type Props = {
  name: string,
  placeholder: string,
  label: string,
  autoFocus?: boolean,
  required?: boolean,
  style?: Object,
  type?: string,
  disabled?: boolean,

  onChange?: () => void,
}

export type State = {
  isFocused: boolean,
  isEmpty: boolean,
}

class BlissInput extends PureComponent<Props, State> {
  static defaultProps = {
    'onChange': () => {},
    'type': 'text',
  }

  constructor(props: ownProps) {
    super(props)
    this.state = {
      'isFocused': false,
      'isEmpty': true,
    }
  }

  onFocus = () => {
    this.setState({
      'isFocused': true,
    })
  }

  onBlur = () => {
    this.setState({
      'isFocused': false,
    })
  }

  onChange = (event: SyntheticInputEvent) => {
    if (event.target.value.length === 0) {
      this.setState({
        'isEmpty': true,
      })
    } else {
      this.setState({
        'isEmpty': false,
      })
    }
    this.props.onChange(event)
  }

  render() {
    return (
      <Label style={this.props.style} htmlFor={this.props.name}>
        <Input
          name={this.props.name}
          className={classNames({
            'is-focused': this.state.isFocused,
            'is-empty': this.state.isEmpty,
          })}
          placeholder={this.props.placeholder}
          onFocus={this.onFocus}
          onBlur={this.onBlur}
          onChange={this.onChange}
          autoFocus={this.props.autoFocus}
          required={this.props.required}
          type={this.props.type || 'text'}
          disabled={this.props.disabled}
        />
        <Span1>
          <Span2>{this.props.label}</Span2>
        </Span1>
      </Label>
    )
  }
}

export default BlissInput
