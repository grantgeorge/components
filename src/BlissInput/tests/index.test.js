import React from 'react'
import { shallow } from 'enzyme'

import BlissInput from '../index'

describe('<BlissInput/>', () => {
  let renderedComponent

  beforeEach(() => {
    renderedComponent = shallow(<BlissInput />)
  })

  it('should render properly', () => {
    expect(renderedComponent).toMatchSnapshot()
  })
})
