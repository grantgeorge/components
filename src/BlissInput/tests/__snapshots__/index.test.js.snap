// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlissInput/> should render properly 1`] = `
<styled.label>
  <styled.input
    className="is-empty"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
  <styled.span>
    <styled.span />
  </styled.span>
</styled.label>
`;
